File: guilib-api/src/main/java/net/developertobi/guilib/api/pagination/GuiPagination.java
package net.developertobi.guilib.api.pagination;

import com.google.common.collect.Multimap;
import net.developertobi.guilib.api.item.GuiItem;
import net.developertobi.guilib.api.item.GuiPos;

import java.util.ArrayList;
import java.util.List;

public interface GuiPagination {

    List<GuiPos> getPositions();
    Multimap<Integer, GuiItem> getItems();

    default List<GuiItem> getPaginationItems() {
        return new ArrayList<>(getItems().values());
    }

    int getLastPageId();
    int getPageAmount();
    int getCurrentPageId();

    boolean isFirstPage();
    boolean isLastPage();

    void page(int pageId);

    void toFirstPage();
    void toLastPage();

    void toNextPage();
    void toPreviousPage();

    void setItemField(int startRow, int startColumn, int endRow, int endColumn);
    void distributeItems(List<GuiItem> items);
    void limitItemsPerPage(int amount);
    void refreshPage();
}

File: guilib-api/bin/main/net/developertobi/inventorylib/api/pagination/GuiPagination.kt
package net.developertobi.inventorylib.api.pagination

import com.google.common.collect.Multimap
import net.developertobi.inventorylib.api.item.GuiItem

interface GuiPagination {

    val positions: List<Any>
    val items: Multimap<Int, GuiItem>

    fun getPaginationItems(): List<GuiItem> = this.items.entries().mapNotNull { it.value }

    fun getLastPageId(): Int
    fun getPageAmount(): Int
    fun getCurrentPageId(): Int

    fun isFirstPage(): Boolean
    fun isLastPage(): Boolean

    fun page(pageId: Int)

    fun toFirstPage()
    fun toLastPage()

    fun toNextPage()
    fun toPreviousPage()

    fun setItemField(startRow: Int, startColumn: Int, endRow: Int, endColumn: Int)
    fun distributeItems(items: List<GuiItem>)
    fun limitItemsPerPage(amount: Int)
    fun refreshPage()

}

